rust/datafusion/src/physical_plan/parquet.rs

```diff
@@ -209,6 +251,479 @@ impl ParquetExec {
+                    min_column_name.as_str(),
+                )?)
+                .and(expr_builder.scalar_expr().lt_eq(rewrite_column_expr(
+                    expr_builder.column_expr(),
+                    expr_builder.column_name(),
+                    max_column_name.as_str(),
+                )?))
+        }
+        Operator::Gt => {
+            let max_column_name = expr_builder.add_max_column();
+            // column > literal => (min, max) > literal => max > literal
+            rewrite_column_expr(
+                expr_builder.column_expr(),
+                expr_builder.column_name(),
+                max_column_name.as_str(),
+            )?
+            .gt(expr_builder.scalar_expr().clone())
+        }
+        Operator::GtEq => {
+            // column >= literal => (min, max) >= literal => max >= literal
+            let max_column_name = expr_builder.add_max_column();
+            rewrite_column_expr(
+                expr_builder.column_expr(),
+                expr_builder.column_name(),
+                max_column_name.as_str(),
+            )?
+            .gt_eq(expr_builder.scalar_expr().clone())
+        }
+        Operator::Lt => {
+            // (input < input) => (predicate)
+            // column < literal => (min, max) < literal => min < literal
+            let min_column_name = expr_builder.add_min_column();
+            rewrite_column_expr(
+                expr_builder.column_expr(),
+                expr_builder.column_name(),
+                min_column_name.as_str(),
+            )?
+            .lt(expr_builder.scalar_expr().clone())
+        }
+        Operator::LtEq => {
+            // (input <= input) => (predicate)
+            // column <= literal => (min, max) <= literal => min <= literal
+            let min_column_name = expr_builder.add_min_column();
+            rewrite_column_expr(
+                expr_builder.column_expr(),
+                expr_builder.column_name(),
+                min_column_name.as_str(),
+            )?
+            .lt_eq(expr_builder.scalar_expr().clone())
+        }
+        // other expressions are not supported
+        _ => {
+            return Ok(expressions::lit(ScalarValue::Boolean(Some(true))));
+        }
+    };
+    expr_builder.build(&statistics_expr)
+}
+
+/// replaces a column with an old name with a new name in an expression
+fn rewrite_column_expr(
+    expr: &Expr,
+    column_old_name: &str,
+    column_new_name: &str,
+) -> Result<Expr> {
+    let expressions = utils::expr_sub_expressions(&expr)?;
+    let expressions = expressions
+        .iter()
+        .map(|e| rewrite_column_expr(e, column_old_name, column_new_name))
+        .collect::<Result<Vec<_>>>()?;
+
+    if let Expr::Column(name) = expr {
+        if name == column_old_name {
+            return Ok(Expr::Column(column_new_name.to_string()));
+        }
+    }
+    utils::rewrite_expression(&expr, &expressions)
+}
+
+#[derive(Debug, Clone, PartialEq)]
```

Comment:
This can use `Copy` too so functions can use `statistics_type: StatisticsType`

Reply:
you are right - I have implemented the `Copy` trait for `StatisticsType` - this makes the code a bit cleaner as it's not necessary to use references all the time